src/internal/EnhancedSwitch.js

```diff
@@ -13,15 +13,15 @@ function getStyles(props, context) {
   return {
     root: {
       position: 'relative',
-      cursor: props.disabled ? 'default' : 'pointer',
+      cursor: props.disabled ? 'not-allowed' : 'pointer',
       overflow: 'visible',
       display: 'table',
       height: 'auto',
       width: '100%',
     },
     input: {
       position: 'absolute',
-      cursor: props.disabled ? 'default' : 'pointer',
+      cursor: props.disabled ? 'not-allowed' : 'pointer',
```

Comment:
@oliviertassinari yes, splitting will be good choice for now. So, I will leave all changes except for EnhancedButton and EnhancedSwitch in this pull-request, correct?

to all: should I close appropriate issue? Or will keep it until cover all components?